operations/helm/charts/mimir-distributed/values.yaml

```diff
@@ -103,6 +103,8 @@ vaultAgent:
   caCertPath: ""
 
 mimir:
+  # -- Adds the appProtocol field to the memberlist service. This allows memberlist to work with istio protocol selection. Ex: "http" or "tcp"
```

Comment:
not sure if this question makes a lot of sense, but can you provide some more details around why this appProtocol is necessary? This might help maintaining this field in the future and potentially help users running not only on istio who are facing similar problems.

Reply:
More about protocol selection for istio https://istio.io/latest/docs/ops/configuration/traffic-management/protocol-selection/ istio needs the appProtocol specified to correctly select the protocol for communication.

You can select default value based on what protocol mimir is using.

Please add this possibility, this becomes a blocker for mimir usage in istio enabled cluster

Reply:
I see. I find it strange that istio cannot just use the value of the existing `protocol` field, especially given that appProtocol will have the same value.

Mimir currently uses only plain TCP over the memberlist port, so technically this option doesn't need to be configurable.

@kendrickclark are you still up to working on this PR or should I take ownership?

Reply:
Sorry for the delay.  I made the suggested changes.

Reply:
I think it's a better idea to hardcode `appProtocol: tcp`. This is the only protocol that mimir uses over the memberlist service. A new value only adds configuration surface and currently doesn't have a use case.

`appProtocl` is stable since k8s 1.20, so adding it to services shouldn't break existing installations. I manually tried to add it to an existing service and it went well. 

If you're ok with doing that you'll also need to run `make build-helm-tests` in the root of the repo and push the changes.

Reply:
I'm fine with that.  Updated the template to reflect this

Reply:
thank you!